src/mono/System.Private.CoreLib/src/System/Threading/LowLevelLock.cs

```diff
@@ -36,5 +36,21 @@ public void VerifyIsLocked()
         {
             Debug.Assert(Monitor.IsEntered(this));
         }
+
+        [Conditional("DEBUG")]
+        public void VerifyIsNotLocked()
+        {
+            Debug.Assert(!Monitor.IsEntered(this));
+        }
+
+#if DEBUG
+        public bool IsLocked
```

Comment:
Where is this used?

Reply:
In `src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.ThreadWaitInfo.Unix.cs`:
`Debug.Assert(s_lock.IsLocked || _waitMonitor.IsLocked);`

Reply:
I only added this after the initial PR when I noticed build errors on CI, since locally I was just testing release (and so this didn't get hit).